build-tools-internal/src/main/groovy/elasticsearch.stable-api.gradle

```diff
@@ -0,0 +1,51 @@
+import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.VersionProperties
+import org.elasticsearch.gradle.internal.BwcVersions
+import org.elasticsearch.gradle.internal.JarApiComparisonTask
+import org.elasticsearch.gradle.internal.info.BuildParams
+
+import java.util.stream.Collectors
+
+import static java.util.Arrays.stream
+
+configurations {
+  stableApi
+}
+
+BuildParams.bwcVersions.allIndexCompatible.findAll({ it.onOrAfter(Version.fromString("8.5.0")) && it != VersionProperties.elasticsearchVersion }).each { bwcVersion ->
+  def baseName = "v${bwcVersion}"
+
+  BwcVersions.UnreleasedVersionInfo unreleasedVersion = BuildParams.bwcVersions.unreleasedInfo(bwcVersion)
+  Configuration stableApiConfiguration = configurations.create("stableApi${baseName}") {
+    // TODO: Temporary workaround for https://github.com/elastic/elasticsearch/issues/73433
+    transitive = false
+  }
+  Object libDependency = null
+
+  if (unreleasedVersion) {
+    // For unreleased snapshot versions, build them from source
+    libDependency = files(project(unreleasedVersion.gradleProjectPath).tasks.named(ext.buildBwcName))
+    // TODO: is there a way to derive this from the project name instead?
+    // libDependency = files(project(unreleasedVersion.gradleProjectPath).tasks.named(buildBwcTaskName(project.name)))
+  } else {
+    // For released versions, download it
+    libDependency = "org.elasticsearch:" + project.name + ":${bwcVersion}"
+  }
+
+  dependencies {
+    "stableApi${baseName}"(libDependency)
+  }
+
+  tasks.register(bwcTaskName(bwcVersion), JarApiComparisonTask) {
+    dependsOn configurations."stableApi${baseName}"
+    // TODO: "normalize" input with a classpath resolution strategy
```

Comment:
I think that's as simple as making `oldJar` a `FileCollection` annotated with `@CompileClasspath`.

Reply:
Thanks, I think I got something that looks a lot cleaner.